src/configure-parameters/configure-parameters.ts

```diff
@@ -0,0 +1,164 @@
+ * keys from the secret JSON object appended.
+ */
+export interface SecretReferenceParameter {
+  key: string
+  secretName: string
+}
+
+export type Parameter =
+  | PlainTextParameter
+  | SecretParameter
+  | SecretReferenceParameter
+
+interface Props {
+  /**
+   * Prefix used for parameter names. Should start with /.
+   */
+  ssmPrefix: string
+  parameters: Parameter[]
+}
+
+export class ConfigureParameters {
+  public parameters: ssm.IParameter[]
+  public ssmPrefix: string
+  private secrets: secretsmanager.ISecret[]
+
+  private configParameters: PlainTextParameter[]
+  private secretParameters: SecretParameter[]
+
+  constructor(scope: cdk.Construct, props: Props) {
+    this.ssmPrefix = props.ssmPrefix
+
+    this.configParameters = []
+    this.secretParameters = []
+
+    for (const parameter of props.parameters) {
+      if ("value" in parameter) {
+        this.configParameters.push(parameter)
+      } else if ("secret" in parameter) {
+        this.secretParameters.push(parameter)
+      } else if ("secretName" in parameter) {
+        this.secretParameters.push({
+          key: parameter.key,
+          secret: secretsmanager.Secret.fromSecretNameV2(
+            scope,
+            `SecretRef${parameter.key}`,
+            parameter.secretName,
+          ),
+        })
+      }
+    }
+
+    const result: ssm.IParameter[] = []
+
+    this.configParameters.forEach((it) => {
+      const param = new ssm.StringParameter(scope, `Config${it.key}`, {
+        type: ssm.ParameterType.STRING,
+        stringValue: it.value,
+        parameterName: `${this.ssmPrefix}/config/${it.key}`,
+      })
+      result.push(param)
+    })
+
+    this.secretParameters.forEach((it) => {
+      const param = new ssm.StringParameter(scope, `Secret${it.key}`, {
+        type: ssm.ParameterType.STRING,
+        stringValue: it.secret.secretArn,
+        parameterName: `${this.ssmPrefix}/secrets/${it.key}`,
+      })
+      result.push(param)
+    })
+
+    this.secrets = this.secretParameters.map((it) => it.secret)
+
+    this.parameters = result
+  }
+
+  grantRead(grantable: iam.IGrantable): void {
+    grantable.grantPrincipal.addToPrincipalPolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["ssm:GetParametersByPath"],
+        resources: [
+          `arn:aws:ssm:${cdk.Stack.of(grantable).region}:${
+            cdk.Stack.of(grantable).account
+          }:parameter${this.ssmPrefix}/*`,
+        ],
+      }),
+    )
+
+    for (const secret of this.secrets) {
+      secret.grantRead(grantable)
+    }
+  }
+
+  /**
+   * Produce a checksum-value that can be used as a kind of
+   * nonce to trigger redeployment on parameter changes.
+   *
+   * Note: If the parameter references a token no change will be visible,
+   * and a manual redeployment might be needed.
+   */
+  get hashValue(): string {
+    const hash = crypto.createHash("md5")
```

Comment:
```suggestion
    const hash = crypto.createHash("sha256")
```

Lurer på om vi kanskje skal gjøre denne endringen like gjerne? Greit å droppe md5 hvis vi kan, selv om vi ikke bryr oss om sikkerhet i dette caset. Det blir en lenger hash men tror ikke vi bryr oss